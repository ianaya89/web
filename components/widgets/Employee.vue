<template>
  <div class="employee-card flex bg-white py-6 px-4 shadow-xl">
    <div class="flex w-1/4 justify-center">
      <avatar v-bind="$props" :classes="'w-16 h-16'" />
    </div>
    <div class="w-3/4 flex flex-col justify-between">
      <div>
        <h5 class="font-medium">
          {{ fullName }}
        </h5>
        <p class="text-grey-l1 -mt-1">
          {{ position }}
        </p>
      </div>

      <a v-if="email" :href="mailTo" class="">{{ email }}</a>
    </div>
  </div>
</template>

<script>
import Avatar from '@/components/Avatar'

export default {
  name: 'Employee',

  components: {
    Avatar
  },

  props: {
    src: {
      type: String,
      default: require('../../assets/images/thumb-fallback.svg')
    },
    fullName: {
      type: String,
      default: 'First name'
    },
    alt: {
      type: String,
      default: 'Image alt'
    },
    email: {
      type: String,
      default: null
    },
    position: {
      type: String,
      required: true
    }
  },

  computed: {
    mailTo() {
      return 'mailto:' + this.email
    }
  }
}
</script>

<style lang="postcss" scoped>
.employee-card {
  min-height: 10rem;
}
</style>
