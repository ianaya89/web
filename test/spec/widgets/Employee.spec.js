import { shallowMount } from '@vue/test-utils'
import Employee from '@/components/widgets/Employee'

describe('Employee', () => {
  it('renders employee without email', () => {
    const wrapper = shallowMount(Employee, {
      propsData: {
        fullName: 'Some name',
        position: 'Founder'
      }
    })

    expect(wrapper.element).toMatchSnapshot()
  })
  it('renders employee with email', () => {
    const wrapper = shallowMount(Employee, {
      propsData: {
        fullName: 'Some name',
        position: 'Founder',
        email: 'test@test.com'
      }
    })

    expect(wrapper.element).toMatchSnapshot()
  })
})
