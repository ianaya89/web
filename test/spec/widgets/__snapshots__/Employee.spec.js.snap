// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Employee renders employee with email 1`] = `
<div
  class="employee-card flex bg-white py-6 px-4 shadow-xl"
>
  <div
    class="flex w-1/4 justify-center"
  >
    <avatar-stub
      alt="Image alt"
      avatarclasses=""
      classes="w-16 h-16"
      email="test@test.com"
      fullname="Some name"
      position="Founder"
      rounded="true"
      src="\\"thumb-fallback.svg\\""
      tooltipcolor="light"
      tooltipposition="top"
    />
  </div>
   
  <div
    class="w-3/4 flex flex-col justify-between"
  >
    <div>
      <h5
        class="font-medium"
      >
        
        Some name
      
      </h5>
       
      <p
        class="text-grey-l1 -mt-1"
      >
        
        Founder
      
      </p>
    </div>
     
    <a
      href="mailto:test@test.com"
    >
      test@test.com
    </a>
  </div>
</div>
`;

exports[`Employee renders employee without email 1`] = `
<div
  class="employee-card flex bg-white py-6 px-4 shadow-xl"
>
  <div
    class="flex w-1/4 justify-center"
  >
    <avatar-stub
      alt="Image alt"
      avatarclasses=""
      classes="w-16 h-16"
      fullname="Some name"
      position="Founder"
      rounded="true"
      src="\\"thumb-fallback.svg\\""
      tooltipcolor="light"
      tooltipposition="top"
    />
  </div>
   
  <div
    class="w-3/4 flex flex-col justify-between"
  >
    <div>
      <h5
        class="font-medium"
      >
        
        Some name
      
      </h5>
       
      <p
        class="text-grey-l1 -mt-1"
      >
        
        Founder
      
      </p>
    </div>
     
    <!---->
  </div>
</div>
`;
